src/client/localplayer.cpp

```diff
@@ -447,18 +447,27 @@ void LocalPlayer::move(f32 dtime, Environment *env, f32 pos_max_d,
 		itemgroup_get(f1.groups, "disable_jump");
 	m_can_jump = ((touching_ground && !is_climbing) || sneak_can_jump) && !m_disable_jump;
 
-	// Jump key pressed while jumping off from a bouncy block
-	if (m_can_jump && control.jump && itemgroup_get(f.groups, "bouncy") &&
-		m_speed.Y >= -0.5f * BS) {
-		float jumpspeed = movement_speed_jump * physics_override_jump;
-		if (m_speed.Y > 1.0f) {
-			// Reduce boost when speed already is high
-			m_speed.Y += jumpspeed / (1.0f + (m_speed.Y / 16.0f));
+	// Jump/Sneak key pressed while bouncing from a bouncy block
+	// Negative bouncy means uncontrollable, so skip in that case.
+	float jumpspeed = movement_speed_jump * physics_override_jump;
+	if (m_can_jump && (control.jump || control.sneak)
+			&& itemgroup_get(f.groups, "bouncy") > 0
+			&& m_speed.Y > 0.0f) {
+		if (!control.jump) {
+			// sneak pressed, but not jump
+			// Subjective testing indicates 1/3 bounce decrease works well.
+			jumpspeed = -m_speed.Y / 3.0f;
 		} else {
-			m_speed.Y += jumpspeed;
+			// jump pressed
+			// Reduce boost when speed already is high
+			jumpspeed = jumpspeed / (1.0f + (m_speed.Y * 3.0f / jumpspeed ));
```

Comment:
Is the difference here vs L1073 intentional? Did the two move codes drift apart enough to need different calculations?

Reply:
This difference is **_very_** intentional.  I was considering adding a comment emphasizing that, I guess I will.

To elaborate: the old move code uses quite fuzzy distance tolerance to determine touching ground, which results (among other things) in each tick being a separate jump until we are high enough to be outside that range.  For normal jumps it doesn't matter so much. But for a trampoline, each jump adds another bounce.

This is why I tweaked the numbers/formula, so the actual **resulting** height using new_move would more-or-less match the previous behavior.

rubenwardy (I believe) mentioned that old move shouldn't be touched at all, but that's not possible here since it would result in trampolines that can ONLY accumulate height.  The options I saw were 1: adjust old_move, or 2: completely _remove_ the old_move code that accumulates height.

I feel it's important to note that the old_move split happened BEFORE the physics changes in 5.3